The vendor has issued an advisory: 
https://kb.netgear.com/000063961/Security-Advisory-for-Authentication-Bypass-Vulnerability-on-the-D7000-and-Some-Routers-PSV-2021-0133

This code checks whether the request line contains todo=PNPX_GetShareFolderList and if it does, it sets need_auth variable to 0. This variable is used to check whether authentication is required to access the resource requested. If it is set to 0, the resource is served without requiring a username and password.

Since it uses strstr to check, the string needs to be present in the original request line, but there is no check to see whether the request line includes anything else.

This means that we can request any file without authentication on the web server, including the configuration file and webpages that disclose the password of the administrative user.

https://ssd-disclosure.com/tag/remote_code_execution/

--- work/mini_httpd-1.30/mini_httpd.c
+++ work/mini_httpd-1.30/mini_httpd.c
@@ -2162,7 +2162,8 @@
 
 #ifdef PNPX
     if (strcasecmp(method_str, get_method_str(METHOD_GET)) == 0 && 
-        strstr(path, "todo=PNPX_GetShareFolderList"))
+        strncmp(path, "/setup.cgi?todo=PNPX_GetShareFolderList", strlen("/setup.cgi?todo=PNPX_GetShareFolderList")) == 0 && 
+        strstr(path, "htm") == NULL )
     {
         need_auth = 0;
     }
